### ISSUE TITLE
Broadcasting with masked array doesn't work with divide (after tile instruction) (Trac #2035)

### ISSUE THREAD

**Issue by numpy-gitbot** [numpy/numpy#2627]
_Original ticket http://projects.scipy.org/numpy/ticket/2035 on 2012-02-02 by trac user elephantterrible, assigned to unknown._

Broadcasting in a division does not work after a tile instruction.

See attached script to reproduce.

If the tile instruction is commented out, the division works.
Division also works if numpy.divide() is used, or if there's no masked value.

Python gives the following error:

```
Traceback (most recent call last):
  File "ma_tiling_issue.py", line 18, in <module>
    a/100. #raises ValueError: invalid return array shape
  File "/usr/local/lib/python2.7/dist-packages/numpy/ma/core.py", line 3654, in __div__
    return divide(self, other)
  File "/usr/local/lib/python2.7/dist-packages/numpy/ma/core.py", line 1078, in __call__
    m |= ma
ValueError: non-broadcastable output operand with shape (3,3,3) doesn't match the broadcast shape (1,3,3,3)
```

The error occurs in all numpy versions (1.5.1, 1.6.1, 2.0.0.dev) with python 2.7.1

The original bug discussion is on numpy-discuss:
http://www.mail-archive.com/numpy-discussion@scipy.org/msg35146.html

**Comment by jjhelmus** [comment]
A similar issue with np.tile modifying the mask of MaskedArrays was reported in #3140

**Comment by cmarmo** [comment]
Hello, I have run the snippet of code and it is not failing anymore: possibly fixed at the same time than #3140?
```py
>>> import numpy as np
>>> np.__version__
'1.24.0.dev0+934.gdb7414b7f'
>>> import numpy.core.numeric as _nx
>>> b=np.ma.array([[1,2,3],[4,5,6]], mask=[[1,0,0],[1,0,0]])
>>> d=4
>>> c = _nx.array(b,copy=True,subok=True,ndmin=d)
>>> b/4
masked_array(
  data=[[--, 0.5, 0.75],
        [--, 1.25, 1.5]],
  mask=[[ True, False, False],
        [ True, False, False]],
  fill_value=1e+20)
>>> c = _nx.array(b,copy=False,subok=True,ndmin=d)
>>> b/4
masked_array(
  data=[[--, 0.5, 0.75],
        [--, 1.25, 1.5]],
  mask=[[ True, False, False],
        [ True, False, False]],
  fill_value=1e+20)
```
Thanks for listening.

**Comment by mattip** [comment]
OK, closing. Thanks for checking.
